cmake/mode.common.cmake

```diff
@@ -145,11 +145,15 @@ macro(update_cxx_flags flags)
   if(NOT DEFINED parsed_args_OPTIMIZATION_LEVEL)
     message(FATAL_ERROR "OPTIMIZATION_LEVEL is missing")
   endif()
-  string(APPEND ${flags}
+  string(TOUPPER ${config} CONFIG)
+  set(cxx_flags "CMAKE_CXX_FLAGS_${CONFIG}")
```

Comment:
I understand the generalization cleanness, but I am not very fond of such settings.
Before the change it was clear at caller site that we set the `CXX` value (e.g. `MAKE_CXX_FLAGS_SANITIZE`)
With this change the caller site sends `Sanitize` that is then upper-cased and appended to `CXX_FLAGS_`
This is much much more complicated to follow/debug in the future

Reply:
thanks for the comment. i agree that it sacrifices the maintainability for flexibility, but i think it's a common practice in CMake world. anyway, i am open to suggestions.